src/contentmanager.cpp

```diff
@@ -864,10 +864,11 @@ void ContentManager::asyncUpdateLibraryFromDir(QString dir)
     });
 }
 
-void ContentManager::handleDisappearedZimFiles(const QStringSet& zimPaths)
+void ContentManager::handleDisappearedZimFiles(const QString& dirPath, const QStringSet& fileNames)
```

Comment:
They actually do, but agree it isn't obvious from the diff log:

`m_knownZimFiles` is set by `setMonitorDirZims`, of which the passed-in value has changed;

`getLibraryZimsFromDir()` is changed, the name didn't appear as it just got truncated in the `src/library.cpp` change

Reply:
@ShaopengLin's analysis is correct. I slightly rebalanced the changes between this commit and a preceding cleanup commit so that the diff is smaller and the reader's attention is not distracted by insignificant changes. I also amended the commit messages with an explanation.